benchmark/gaia/gaia_benchmark.py

```diff
@@ -0,0 +1,203 @@
+import os, json, re, string, warnings
+from camel.agents import ChatAgent
+from typing import Literal, List, Union
+from camel.messages import BaseMessage
+import time
+SCRIPT_PATH = os.path.realpath(__file__)
+SCRIPT_DIR = os.path.dirname(SCRIPT_PATH)
+DATASET_DIR = os.path.join(SCRIPT_DIR, "Dataset")
+if not os.path.isdir(DATASET_DIR):
+    os.mkdir(DATASET_DIR)
+class GAIABenchmark:
+    r"""
+    Please using huggingface-cli login to get authorization
+    From Hugging Face download GAIA dataset
+    This Class will create a folder to cache GAIA dataset
+    """
+    def __init__(self) -> None:
+        self.validation_tasks = [[], [], []]
+        self.test_tasks = [[], [], []]
+
+    def download(self) -> None:
+        r"""
+        download GAIA dataset
+        """
+        from huggingface_hub import snapshot_download
+        snapshot_download(
+            repo_id="gaia-benchmark/GAIA",
+            repo_type="dataset",
+            local_dir= DATASET_DIR,
+            local_dir_use_symlinks=True,
+        )
+        validation = os.path.join(DATASET_DIR, "2023", "validation")
+        test = os.path.join(DATASET_DIR, "2023", "test")
+        with open(os.path.join(validation, "metadata.jsonl")) as f:
+            for line in f:
+                data = json.loads(line)
+                self.validation_tasks[data["Level"] - 1].append(data)
+        with open(os.path.join(test, "metadata.jsonl")) as f:
+            for line in f:
+                data = json.loads(line)
+                if data["task_id"] == "0-0-0-0-0":
+                    continue
+                self.test_tasks[data["Level"] - 1].append(data)
+
+    def eval(self, 
+            agent: ChatAgent,
+            val_or_test : Literal['validation', 'test'] = "validation", 
+            level: Union[List[Literal[1, 2, 3]], Literal["all"]] = [1], 
+            ) -> float:
+        results = []
+        scores = []
+        if val_or_test == "validation":
+            tasks = self.validation_tasks
+        elif val_or_test == "test":
+            tasks = self.test_tasks
+        
+        levels = [0, 1, 2] if level == "all" else [level - 1]
+        for lvl in levels:
+            for task in tasks[lvl]:
+                final_answer = task['Final answer']
+                user_msg = BaseMessage.make_user_message(
+                    role_name="User",
+                    content=task['Question'],
+                )
+                retries = 0
+                max_retries = 5
+                initial_backoff = 1
+                max_backoff = 100
+                backoff = initial_backoff
```

Comment:
I don't think that we need the retries. It is just hundreds of questions, it should be fine.